plugins/autodim.koplugin/main.lua

```diff
@@ -0,0 +1,257 @@
+                        unit = _("m"),
+                        precision = "%0.1f",
+                        ok_always_enabled = true,
+                        callback = function(spin)
+                            if not spin then return end
+                            self.autodim_starttime_m = spin.value
+                            G_reader_settings:saveSetting("autodim_starttime_minutes", spin.value)
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                        extra_text = _("Disable"),
+                        extra_callback = function()
+                            self.autodim_starttime_m = -1
+                            G_reader_settings:saveSetting("autodim_starttime_minutes", -1)
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                    }
+                    UIManager:show(idle_dialog)
+                    if touchmenu_instance then touchmenu_instance:updateItems() end
+                end,
+                keep_menu_open = true,
+            },
+            {
+                text_func = function()
+                    return T(_("Dimmer duration: %1 s"), self.autodim_duration_s)
+                end,
+                enabled_func = function() return self.autodim_starttime_m > 0 end,
+                callback = function(touchmenu_instance)
+                    local dimmer_dialog = SpinWidget:new{
+                        title_text = _("Automatic dimmer duration"),
+                        info_text = _("Delay to reach the lowest brightness."),
+                        value = self.autodim_duration_s,
+                        default_value = DEFAULT_AUTODIM_DURATION_S,
+                        value_min = 0,
+                        value_max = 300,
+                        value_step = 1,
+                        value_hold_step = 10,
+                        precision = "%1d",
+                        unit = _("s"),
+                        callback = function(spin)
+                            if not spin then return end
+                            self.autodim_duration_s = spin.value
+                            G_reader_settings:saveSetting("autodim_duration_seconds", spin.value)
+                            self:_schedule_autodim_task()
+                            if touchmenu_instance then touchmenu_instance:updateItems() end
+                        end,
+                    }
+                    UIManager:show(dimmer_dialog)
+                    if touchmenu_instance then touchmenu_instance:updateItems() end
+                end,
+                keep_menu_open = true,
+            },
+            {
+                text_func = function()
+                    return T(_("Lowest fraction of brightness: %1 %"), self.autodim_fraction)
```

Comment:
Is a "lowest fraction" a thing? Shouldn't that be the smallest fraction?

Reply:
Which circles back to not really liking fraction in this context. `Lowest` is a clearer word in any case.

Reply:
Circle back: What about `Dim to 20% of the reading brightness`. 
With this statement everything is said: "Dim to 20%" says 20% in respect of reading brightness. And  "reading brightness" itself says what the base for the 20% is.

Reply:
Something along those lines seems better to me. Is _reading brightness_ really better than, say, _regular brightness_?

Reply:
Yeah, `regular` sounds good ;).